### DISCUSSION TITLE
Add legend to a map

### DISCUSSION THREAD

**Discussion by PietroMart** [mtex-toolbox/mtex#2050] [Ask Anything]
Hello, I am using the Martensite Parent Grain Reconstructor in the MTEX tool and I am wondering how couldn't I plot the legend? I am displaying the map with the variants of each martensite block colored differently and I want to display a legend to know which color represents which variant. How can I? If I use the legend command I only get a blank rectangle, with no labels inside, and the same if I pass labels as argument of legend

**Comment by ralfHielscher** [comment]
Have a look [here](https://mtex-toolbox.github.io/Legends.html). You need to pass the option `'DisplayName'` to the plotting command in order that the corresponding object appears in the legend, i.e.,
``` matlab
plot(ebsd(variant==1),'FaceColor','blue','DisplayName','Variant 1')
```
Ralf.
